src/xGadgetron/pGadgetron/Gadgetron.py

```diff
@@ -292,6 +293,28 @@ def patient_table_position(self):
     def info(self, method):
         return eval('self.' + method + '()')
 
+    @property
+    def shape(self):
+        return self.matrix_size()
+
+    @property
+    def __array_interface__(self):
+        """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
+        dt = self.data_type()
+        typestr = ['<i2', '<i2', '<i4', '<i4', '<f4', '<f8', '<c8', '<c16']
+        return {'shape': self.shape, 'typestr': typestr[dt - 1], 'version': 3,
+                'data': (parms.size_t_par(self.handle, 'Image', 'address'), False)}
+
+    def asarray(self, xp=numpy, copy=None, **kwargs):
```

Comment:
can't we use https://github.com/SyneRBI/SIRF/blob/8fbfe5294e4efd57d12ae15d87b97f64e96d09d4/src/common/SIRF.py#L237

Reply:
no, we cannot - `Image` is not `DataContainer`

Reply:
but they could both inherit from the same ABC

Reply:
> no, we cannot - `Image` is not `DataContainer`

I see. I guess `ImageData` is, hence you don't have `asarray` there explicitly. I guess I've lost track by now why we have 2 different classes.

Maybe not worth creating another ABC (parent of `DataContainer` I guess) with only `asarray` in there, unless other stuff could be as well.